core/src/main/java/com/yahoo/ycsb/Client.java

```diff
@@ -343,7 +346,12 @@ public void run()
         while (((_opcount == 0) || (_opsdone < _opcount)) && !_workload.isStopRequested())
         {
 
-          if (!_workload.doInsert(_db,_workloadstate))
+          if (_dodelete)
+          {
+            _workload.doDelete(_db,_workloadstate);
```

Comment:
Should track the success failure of operations, somehow. Not sure breaking like the failure of doInsert() is the right thing here since we are likely trying to cleanup after a workload that may delete records.